src/values/ObjectValue.js

```diff
@@ -481,6 +491,21 @@ export default class ObjectValue extends ConcreteValue {
     return keyArray;
   }
 
+  getSnapshot(): AbstractObjectValue {
```

Comment:
This doesn't deal with symbols, prototypes, or anything except string properties. Should it be renamed to something more specific like `getPropertiesSnapshot`?

Reply:
I don't think so, what we are really getting here is a name represents the state of the object at this temporal point. Right now it does not deal with prototypes (and I don't think it is needed for they way Object.assign is currently implemented), but eventually it should. I don't foresee us wanting to have both getPropertiesSnapshot and getSnapshot.

Reply:
This should at least have a big comment clarifying this then. It's sometimes hard to tell if it is safe to use a method, when adding a new feature, if it is partially implemented.